toolkit/tools/pkg/imagecustomizerlib/imagefileinfo.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/shell"
+)
+
+type ImageFileInfo struct {
+	Format      string `json:"format"`
+	VirtualSize int64  `json:"virtual-size"`
+}
+
+func getImageFileInfo(inputImageFile string) (ImageFileInfo, error) {
+	stdout, _, err := shell.Execute("qemu-img", "info", "--output", "json", inputImageFile)
+	if err != nil {
+		return ImageFileInfo{}, fmt.Errorf("failed to check image file's disk format:\n%w", err)
+	}
+
+	info := ImageFileInfo{}
+	err = json.Unmarshal([]byte(stdout), &info)
```

Comment:
`err = json.Unmarshal(stdout, &info)`? Is `stdout` already a string?

Reply:
`stdout` is a `string`. But `json.Unmarshal` wants a byte array.